sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/PlannerFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.impl;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import org.apache.calcite.tools.FrameworkConfig;
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.Planner;
+
+/** Factory that creates an instance of {@link Planner}. */
+class PlannerFactory {
+  public static Planner create(FrameworkConfig config, BeamSqlPipelineOptions options) {
+    if (options
+        .getPlannerImplClassName()
+        .equals(BeamSqlConstants.DEFAULT_BEAM_SQL_PLANNER_IMPL_CLASS_NAME)) {
```

Comment:
I think that if you don't provide the default value in the pipelineoptions, then it would be simpler here, e.g. only use reflection when the value is not null

Reply:
I think it helps understand Beamsql code if exposing what's the default planner.